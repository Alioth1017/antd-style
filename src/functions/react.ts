import _styled from '@emotion/styled';

import type { CreateStyled } from '@/types';

export const styled = _styled as CreateStyled;
export { css as reactCss, ThemeProvider as EmotionThemeProvider, withTheme } from '@emotion/react';
