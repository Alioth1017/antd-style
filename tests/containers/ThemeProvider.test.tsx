import { render } from '@testing-library/react';

import { css, ThemeProvider, useTheme } from 'antd-style';

interface TestDesignToken {
  customBrandColor: string;
}
interface TestDesignStylish {
  defaultText: string;
}

declare module 'antd-style' {
  // eslint-disable-next-line @typescript-eslint/no-empty-interface
  interface CustomToken extends TestDesignToken {}

  // eslint-disable-next-line @typescript-eslint/no-empty-interface
  interface CustomStylish extends TestDesignStylish {}
}

describe('ThemeProvider', () => {
  it('自定义 Token', () => {
    const App = () => {
      const theme = useTheme();
      return <div style={{ color: theme.customBrandColor }}>{theme.customBrandColor}</div>;
    };

    const { container } = render(
      <ThemeProvider<TestDesignToken> customToken={{ customBrandColor: '#c956df' }}>
        <App />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
    expect(container.firstChild).toHaveStyle({ color: '#c956df' });
  });

  it('注入自定义 Stylish', () => {
    const App = () => {
      const theme = useTheme();
      return <div className={theme.stylish.defaultText}>普通文本</div>;
    };

    const { container } = render(
      <ThemeProvider<any, TestDesignStylish>
        customStylish={{
          defaultText: css`
            font-size: 14px;
            font-weight: 500;
            color: #333;
          `,
        }}
      >
        <App />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
    expect(container.firstChild).toHaveStyle({ color: '#333', fontSize: 14, fontWeight: 500 });
  });
});
