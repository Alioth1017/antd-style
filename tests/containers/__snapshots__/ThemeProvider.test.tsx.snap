// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeProvider 注入自定义 Stylish 1`] = `
.emotion-0 {
  font-size: 14px;
  font-weight: 500;
  color: #333;
}

<div>
  <div
    class="emotion-0"
  >
    普通文本
  </div>
</div>
`;

exports[`ThemeProvider 自定义 Token 1`] = `
<div>
  <div
    style="color: rgb(201, 86, 223);"
  >
    #c956df
  </div>
</div>
`;
